packages/bot/src/stores/run-panel-store.js

```diff
@@ -68,10 +70,26 @@ export default class RunPanelStore {
             return;
         }
 
-        this.registerBotListeners();
+        const top_blocks = Blockly.derivWorkspace.getTopBlocks();
```

Comment:
@sam-binary can you keep these in blockly? its better to keep the code in one place if possible? for instance add a isvaild to `Blockly` and check the rules there